revops/revops.js

```diff
@@ -45,4 +47,38 @@ async function populateServices() {
   container.appendChild(servicesList);
 }
 
+async function populateBlogPosts() {
+  const request = new Request("blog-posts.json");
+
+  const response = await fetch(request);
+  blogPosts = await response.json();
+}
+
+async function getLatestBlogPosts() {
+  await populateBlogPosts();
+
+  const blogPostsList = document.getElementById("recent-blog-posts");
+  const latestBlogPost = document.getElementById("post-preview");
+
+  blogPosts.forEach((post) => {
+    const clone = latestBlogPost.content.cloneNode(true);
+
+    const blogPostLink = clone.getElementById("post-preview-link");
+    blogPostLink.href = post.Slug;
+
+    const blogPostThumbnail = clone.getElementById("post-preview-thumbnail");
+    blogPostThumbnail.src = post["Thumbnail image"];
+
+    const blogPostTitle = clone.getElementById("post-preview-title");
+    blogPostTitle.innerHTML = post.Name;
+
+    const blogPostSummary = clone.getElementById("post-preview-summary");
+    blogPostSummary.innerHTML = post["Post Summary"];
+
+    blogPostsList.appendChild(clone);
```

Comment:
C.f. other note on how to generate HTML in native Javascript.

But this makes me thing something else: instead of fetching the data in the browser when the page loads, you can just generate a page per blog post. C.f. how it is done in Jekyll: https://jekyllrb.com/docs/collections/

Reply:
I have now rebuilt the services and blog posts structures using Jekyll collections